BUG: false positives in B030

```python
# reprod.py
black_list = (TypeError, ValueError)
grey_list = (ImportError, FileNotFoundError)

try:
    pass
except (*grey_list, *black_list):
    pass
```
```
$ flake8 reprod.py --select B030 # no error reported !
$ ruff reprod.py --select B030 --isolated
reprod.py:6:9: B030 `except` handlers should only be exception classes or tuples of exception classes
reprod.py:6:21: B030 `except` handlers should only be exception classes or tuples of exception classes
Found 2 errors.
```

`ruff --version` -> `ruff 0.0.260`

Note that this edge case used to be problematic in flake8-bugbear too (see https://github.com/PyCQA/flake8-bugbear/issues/153)

--- a/crates/ruff/resources/test/fixtures/flake8_bugbear/B030.py
+++ b/crates/ruff/resources/test/fixtures/flake8_bugbear/B030.py
@@ -30,6 +30,11 @@ except (ValueError, *(RuntimeError, (KeyError, TypeError))):  # error
 
 try:
     pass
+except (*a, *(RuntimeError, (KeyError, TypeError))):  # error
+    pass
+
+try:
+    pass
 except (ValueError, *(RuntimeError, TypeError)):  # ok
     pass
 
@@ -38,8 +43,34 @@ try:
 except (ValueError, *[RuntimeError, *(TypeError,)]):  # ok
     pass
 
+
+try:
+    pass
+except (*a, *b):  # ok
+    pass
+
+
+try:
+    pass
+except (*a, *(RuntimeError, TypeError)):  # ok
+    pass
+
+
+try:
+    pass
+except (*a, *(b, c)):  # ok
+    pass
+
+
+try:
+    pass
+except (*a, *(*b, *c)):  # ok
+    pass
+
+
 def what_to_catch():
     return ...
+
 
 try:
     pass

--- a/crates/ruff/src/rules/flake8_bugbear/rules/except_with_empty_tuple.rs
+++ b/crates/ruff/src/rules/flake8_bugbear/rules/except_with_empty_tuple.rs
@@ -13,17 +13,17 @@ pub struct ExceptWithEmptyTuple;
 impl Violation for ExceptWithEmptyTuple {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Using except (): with an empty tuple does not handle/catch anything. Add exceptions to handle.")
+        format!("Using `except ():` with an empty tuple does not catch anything; add exceptions to handle")
     }
 }
 
 /// B029
 pub fn except_with_empty_tuple(checker: &mut Checker, excepthandler: &Excepthandler) {
     let ExcepthandlerKind::ExceptHandler { type_, .. } = &excepthandler.node;
-    if type_.is_none() {
+    let Some(type_) = type_ else {
         return;
-    }
-    let ExprKind::Tuple { elts, .. } = &type_.as_ref().unwrap().node else {
+    };
+    let ExprKind::Tuple { elts, .. } = &type_.node else {
         return;
     };
     if elts.is_empty() {

--- a/crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs
+++ b/crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs
@@ -33,7 +33,7 @@ fn flatten_starred_iterables(expr: &Expr) -> Vec<&Expr> {
                 ExprKind::Tuple { elts, .. } | ExprKind::List { elts, .. } => {
                     exprs_to_process.append(&mut elts.iter().collect());
                 }
-                _ => flattened_exprs.push(expr),
+                _ => flattened_exprs.push(value),
             },
             _ => flattened_exprs.push(expr),
         }
@@ -48,12 +48,17 @@ pub fn except_with_non_exception_classes(checker: &mut Checker, excepthandler: &
         return;
     };
     for expr in flatten_starred_iterables(type_) {
-        match expr.node {
-            ExprKind::Attribute { .. } | ExprKind::Name { .. } | ExprKind::Call { .. } => (),
-            _ => checker.diagnostics.push(Diagnostic::new(
+        if !matches!(
+            &expr.node,
+            ExprKind::Subscript { .. }
+                | ExprKind::Attribute { .. }
+                | ExprKind::Name { .. }
+                | ExprKind::Call { .. },
+        ) {
+            checker.diagnostics.push(Diagnostic::new(
                 ExceptWithNonExceptionClasses,
                 Range::from(expr),
-            )),
+            ));
         }
     }
 }

